src/vue-components/infl-footnote.vue

```diff
@@ -63,7 +66,39 @@
 
         target.setAttribute('data-x', x)
         target.setAttribute('data-y', y)
+      },
+
+      isOutOfRightXBound(childBR, parentBR) {
+        return (childBR.x + childBR.width) > (parentBR.x + parentBR.width)
+      },
+
+      isOutOfLeftXBound(childBR, parentBR) {
+        return childBR.x < 0
+      },
+
+      checkBounds () {
+        let inflpopup = this.$el.querySelector('.alpheios-inflections__footnote-popup')
+        let inflpanel = this.$el.closest('#alpheios-panel__inflections-panel')      
+
+        let popupBR = inflpopup.getBoundingClientRect()
+        let panelBR = inflpanel.getBoundingClientRect()
+
+        if (this.isOutOfRightXBound(popupBR, panelBR)) {
+          this.popupAlignmentStyles.transform = 'translateX(-100%)'
```

Comment:
If I understand correctly, this should shift it by popup's width so that the top right corner of the popup would be over the footnote link. I'm thinking maybe we should rather shift it to the left just enough so that it won't stick out of the panel (i.e. so that the right border of the popup be near the right panel border)? I think this will be more resembling its "normal" position with center of the popup over the footnote link and would be a less drastic change in position. 

To do that we can add `calc` to the transform to adjust it by how much a popup sticks out of the panel or maybe simply use `right: 0` CSS rule. 

What do you think?

Reply:
@kirlat, may be I didn't understand you correctly.

Now it simply calcs from the parent infl-footnote link - 3 ways:
1) it is in the middle above the footlink
2) it is at the left border of the footnote link
3) it is at the right border of the footnote link.

This is how transformX works.
I don't think that it is really a good idea to apply both right (left) and transformX, because they are dependent on each other.

Do you think that it is needed to be not exactly at the right border or left?
D you think that it is best to remove transformX and try to calculate in px the current position according to panel border?
Or you suggest to use not -100% but -80% in transformX (with checking the outbounding)?

Reply:
@irina060981, what I want to suggest is this.

Let's say a popup is at the right side of the panel and it does not fit in completely into it. Then with the current code it will be adjusted to be shown like this:
![image](https://user-images.githubusercontent.com/18631055/45427172-42bcc680-b6af-11e8-8ba7-93ade63bbb64.png)

I think it might be better to show it like below:
![image](https://user-images.githubusercontent.com/18631055/45427286-86afcb80-b6af-11e8-86f4-71bda46a2764.png)
so that the right side of the popup will touch the right side of the panel.

My suggestion to achieve this was to use either:
a. Adjust current `transform: translateX(-50%)` rule to something like `transform: translateX(calc(-50% - Dpx))`, where `Dpx` is a distance a popup would stick out of the panel if placed the "normal" way (please see image below):
![image](https://user-images.githubusercontent.com/18631055/45427775-c2976080-b6b0-11e8-8147-0a7297711bf4.png)
`isOutOfRightXBound()` can be modified to return that distance instead of the boolean.

b. Cancel transform rule by setting it to `none` and user `right: 0` for positioning instead.

I think this way it will look more natural and closer to the way we show a popup in a "normal" position. @balmas, what do you think?

Reply:
yes I think that would be fine. I don't know if we can get this perfect though, so I am not in favor of spending a huge amount of time on it.